src/WebJobs.Script/Binding/FunctionBinding.cs

```diff
@@ -295,5 +326,18 @@ public static void ConvertStreamToValue(Stream stream, DataType dataType, ref ob
                     break;
             }
         }
+
+        internal static string ToJson(ExpandoObject value)
+        {
+            return JsonConvert.SerializeObject(value, Formatting.None, _expandoObjectJsonConverter);
+        }
+
+        internal static JObject ToJObject(ExpandoObject value)
+        {
+            // TODO: we should optimize this to remove the double
+            // serialization/deserialization
+            string json = ToJson(value);
```

Comment:
Not sure of an easier out of the box way to recursively convert from an ExpandoObject to a JObject.